src/main/java/com/codeborne/selenide/proxy/SelenideProxyServer.java

```diff
@@ -116,15 +116,14 @@ public boolean isStarted() {
    * @param requestFilter the filter
    */
   public void addRequestFilter(String name, RequestFilter requestFilter) {
-    if (isRequestFilterAdded(name)) {
+    RequestFilter currentFilter = requestFilter(name);
+    if (currentFilter != null && !currentFilter.equals(requestFilter)) {
```

Comment:
Maybe better to use such verification and skip next `currentFilter` null check as well?
```suggestion
    if (Objects.equals(currentFilter, requestFilter)) {
```

Reply:
@rosolko Nice try, but no :)
Then this method will throw exception if `currentFilter == null`. But should not.